tools/autograd/gen_variable_type.py

```diff
@@ -137,7 +137,8 @@
 # If tensor_name == out_tensor_name, used to enforce (1), otherwise used for (2)
 ENFORCE_SAME_TENSOR_STORAGE = CodeTemplate("""\
 if (${tensor_name}_storage_saved.has_value())
-  AT_ASSERT(${tensor_name}_storage_saved.value().is_alias_of(${out_tensor_name}.storage()));
+  // AT_ASSERT(${tensor_name}_storage_saved.value().is_alias_of(${out_tensor_name}.storage()));
```

Comment:
The new test is not valid because you won't detect if the input was actually modified inplace. Because you read it after it was (potentially) modified.

Reply:
Deleted in the new PR. In particular, I added a new `FunctionalStorageImpl` so calls to `a.storage().is_alias_of(b.storage())` is always valid.